lib/Parse/ParseType.cpp

```diff
@@ -1391,10 +1388,8 @@ bool Parser::isImplicitlyUnwrappedOptionalToken(const Token &T) const {
   // implicitly unwrapped optional too. We'll munch off the '!', so long as it
   // is left-bound with the type (i.e., parsed as a postfix or unspaced binary
   // operator).
-  if ((T.is(tok::oper_postfix) || T.is(tok::oper_binary_unspaced)) &&
-      T.getText().startswith("!"))
-    return true;
-  return false;
+  return ((T.is(tok::oper_postfix) || T.is(tok::oper_binary_unspaced)) &&
```

Comment:
Same problem here as above. We can really improve the readability of these lines by splitting the condition instead of just inlining it into the return.